py-polars/polars/_utils/pycapsule.py

```diff
@@ -28,11 +29,13 @@ def pycapsule_to_frame(
 ) -> DataFrame:
     """Convert PyCapsule object to DataFrame."""
     if hasattr(obj, "__arrow_c_array__"):
-        # This uses the fact that PySeries.from_arrow_c_array will create a
-        # struct-typed Series. Then we unpack that to a DataFrame.
-        tmp_col_name = ""
         s = wrap_s(PySeries.from_arrow_c_array(obj))
-        df = s.to_frame(tmp_col_name).unnest(tmp_col_name)
+        if isinstance(obj, pa.RecordBatch):
+            # Maintain historical behavior of unnesting RecordBatch
+            tmp_col_name = ""
+            df = s.to_frame(tmp_col_name).unnest(tmp_col_name)
```

Comment:
Why do it this way?
The ArrayStream that corresponds to the DataFrame should be of type Struct, so it should always be possible to unnest it.